linode/resource_linode_vlan_test.go

```diff
@@ -0,0 +1,200 @@
+package linode
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/linode/linodego"
+)
+
+const (
+	testVLANResName     = "linode_vlan.test"
+	testVLANDescription = "terraform-provider-linode acctest"
+)
+
+func init() {
+	resource.AddTestSweepers("linode_vlan", &resource.Sweeper{
+		Name: "linode_vlan",
+		F:    testSweepLinodeVLAN,
+	})
+}
+
+func testSweepLinodeVLAN(prefix string) error {
+	client, err := getClientForSweepers()
+	if err != nil {
+		return fmt.Errorf("failed to get client: %s", err)
+	}
+
+	vlan, err := client.ListVLANs(context.Background(), nil)
+	if err != nil {
+		return fmt.Errorf("failed to get VLANs: %s", err)
+	}
+	for _, vlan := range vlan {
+		if vlan.Description != testVLANDescription {
+			continue
+		}
+		if err := client.DeleteVLAN(context.Background(), vlan.ID); err != nil {
+			return fmt.Errorf("failed to destroy VLAN %d during sweep: %s", vlan.ID, err)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckLinodeVLANAttachedLinodes(linodes ...*linodego.Instance) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
```

Comment:
I think you just want to make sure the set of expected equals the set of actual.